core/src/main/scala/quasar/destination/s3/S3DestinationModule.scala

```diff
@@ -54,6 +57,17 @@ object S3DestinationModule extends DestinationModule {
     case Right(cfg) => cfg.copy(credentials = RedactedCreds).asJson
   }
 
+  def unapplyBucketUri(config: Json)(bucketUri: BucketUri): Either[InitializationError[Json], (Option[URI], Bucket)] =
+    (for {
```

Comment:
Redundand `(` and `)`. 

Also, why is this `unapply` there is no `apply` which takes `URI` and `Bucket` and provide `BucketUri`, right?

Reply:
The bracket that's associated with this one `(for` is
```
    } yield (endpoint, Bucket(bucket))).toRight(DestinationError.malformedConfiguration((destinationType, config, s"Could not obtain bucket from bucket URI ${bucketUri.value}")))
                                      ^
```
This pair is necessary to make it compile. Or do you mean another one?

> Also, why is this `unapply`

I was looking for a good name and choose this because of it's a bit similar to extractors. It's not exactly an extractor but that's why I also named it a bit different than the exact `unapply` :).
I also considered `extractFromBucketUri`. Happy to change it into this or take any other suggestion if you don't like either.

Reply:
Hm... I meant that `(for` actually. 

`extractFromBucketUri` has its own drawbacks. :+1: for `unapply` as it's common name for destructors